pkg/render/applicationlayer/applicationlayer.go

```diff
@@ -159,6 +159,11 @@ func (c *component) Objects() ([]client.Object, []client.Object) {
 		objs = append(objs, c.securityContextConstraints())
 	}
 
+	// Delete all objects if both log collection and WAF are disabled
+	if !c.config.LogsEnabled && !c.config.WAFEnabled {
```

Comment:
Why do we need this? Deletion should be handled by owner references, and I thought that we were validating against this being a possible configuration (i.e., it should not be possible to create an ApplicationLayer resource with both of these disabled, because that has no meaning whatsoever)

Reply:
We do - in case when e.g. either L7 or WAF was enabled and later on disabled the Envoy daemonset would stay present. We don't want that. This comes from manual testing.

Reply:
> was enabled and later on disabled 

But this is the problem - it should not be possible to create an Application layer resource with both `LogsEnabled: false` and `WAFEnabled: false` - that's a nonsensical API request, it's basically saying "Here's an explicit request to do nothing".

We should be enfocing users deleted the ApplicationLayer resource if they want to get rid of the envoy daemonset  - that's how every other one of our APIs works.

Reply:
Done as per our conversation. This won't be needed now.